lib/avatarDialog/avatar_dialog.dart

```diff
@@ -28,7 +27,7 @@ class _AvatarDialogState extends State<AvatarDialog> {
   @override
```

Comment:
in _scrollToIndex function  after scrolling , you are setting the index. Thats why the check box comes after scroll.
Remove setting index. This circle backs to why 2 variables are being used to store index.